diff --git a/node_modules/expo-camera/android/src/main/java/expo/modules/camera/ExpoCameraView.kt b/node_modules/expo-camera/android/src/main/java/expo/modules/camera/ExpoCameraView.kt
index b3591cd..a7c874a 100644
--- a/node_modules/expo-camera/android/src/main/java/expo/modules/camera/ExpoCameraView.kt
+++ b/node_modules/expo-camera/android/src/main/java/expo/modules/camera/ExpoCameraView.kt
@@ -7,6 +7,7 @@ import android.graphics.SurfaceTexture
 import android.net.Uri
 import android.os.Bundle
 import android.view.View
+import android.util.Log
 import com.google.android.cameraview.CameraView
 import expo.modules.camera.CameraViewHelper.getCamcorderProfile
 import expo.modules.camera.CameraViewHelper.getCorrectCameraRotation
@@ -314,13 +315,19 @@ class ExpoCameraView(
     }
   }

-  override fun onHostPause() {
-    if (!isPaused && cameraView.isCameraOpened) {
-      faceDetector?.release()
-      isPaused = true
-      cameraView.stop()
-    }
-  }
+ override fun onHostPause() {
+     if (!isPaused && cameraView?.isCameraOpened == true) {
+         try {
+            faceDetector?.release()
+            cameraView?.stop()
+         } catch (e: Exception) {
+            Log.e("onHostPause", "Error occurred: ${e.message}")
+         } finally {
+             isPaused = true
+         }
+     }
+ }
+

   override fun onHostDestroy() {
     faceDetector?.release()
