diff --git a/node_modules/expo-face-detector/ios/EXFaceDetector.podspec b/node_modules/expo-face-detector/ios/EXFaceDetector.podspec
index ed661e3..9cf6103 100644
--- a/node_modules/expo-face-detector/ios/EXFaceDetector.podspec
+++ b/node_modules/expo-face-detector/ios/EXFaceDetector.podspec
@@ -19,10 +19,10 @@ Pod::Spec.new do |s|
   # even though `GoogleMLKit/FaceDetection` depends on all `MLKit*` references below
   # framework generation code (prebuilds) cannot locate them properly, so these are defined explicitly
   # TODO: research why xcodegen fails to detect dependencies of dependencies (resulted .xcodeproj is missing them)
-  s.dependency 'GoogleMLKit/FaceDetection', '2.6.0'
-  s.dependency 'MLKitFaceDetection', '1.5.0'
-  s.dependency 'MLKitCommon', '5.0.0'
-  s.dependency 'MLKitVision', '3.0.0'
+  s.dependency 'GoogleMLKit/FaceDetection', '4.0.0'
+  s.dependency 'MLKitFaceDetection', '3.0.0'
+  s.dependency 'MLKitCommon', '9.0.0'
+  s.dependency 'MLKitVision', '5.0.0'

   if !$ExpoUseSources&.include?(package['name']) && ENV['EXPO_USE_SOURCE'].to_i == 0 && File.exist?("#{s.name}.xcframework") && Gem::Version.new(Pod::VERSION) >= Gem::Version.new('1.10.0')
     s.source_files = "#{s.name}/**/*.h"
